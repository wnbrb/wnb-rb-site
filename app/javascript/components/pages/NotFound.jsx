import React from 'react';
import { Helmet } from 'react-helmet-async';
import SharedLayout from 'components/layout/SharedLayout';
import NewLogo from 'components/icons/NewLogo';

import 'stylesheets/not_found';

const NotFound = () => (
    <>
        <Helmet>
            <title>WNB.rb: 404</title>
        </Helmet>

        <SharedLayout>
            <div className="not-found-container">
                <NewLogo className="h-60 w-60" />
                <div className="not-found-text-container">
                    <h1 className="title-text-lg">404</h1>
                    <p>We couldn&apos;t find the page you were looking for.</p>
                </div>
            </div>
        </SharedLayout>
    </>
);

export default NotFound;
