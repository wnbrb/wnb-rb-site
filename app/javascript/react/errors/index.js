export class UnauthorizedError extends Error {
    constructor(message) {
        super(message);
        this.name = 'UnauthorizedError';
    }
}
