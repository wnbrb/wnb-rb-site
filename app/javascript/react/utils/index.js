const MS_IN_DAY = 24 * 60 * 60 * 1000;

export const postedAtString = (createdAtString) => {
    const today = new Date();
    const postedDate = new Date(createdAtString);

    const diff = (+today - +postedDate) / MS_IN_DAY;

    if (diff <= 1) {
        return 'today';
    }

    if (diff < 7) {
        return `${Math.floor(diff)} days ago`;
    }

    if (diff < 30) {
        const weekDiff = Math.floor(diff / 7);
        const weekOrWeeks = weekDiff === 1 ? 'week' : 'weeks';

        return `${weekDiff} ${weekOrWeeks} ago`;
    }

    const monthDiff = Math.floor(diff / 30);
    const monthOrMonths = monthDiff === 1 ? 'month' : 'months';

    return `${monthDiff} ${monthOrMonths} ago`;
};
