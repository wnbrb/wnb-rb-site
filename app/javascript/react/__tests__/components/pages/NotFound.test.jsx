import React from 'react';
import { render } from '@testing-library/react';
import { HelmetProvider } from 'react-helmet-async';
import NotFound from '../../../components/pages/NotFound';

// Mock SharedLayout component
jest.mock('../../../components/layout/SharedLayout', () => {
    // eslint-disable-next-line react/prop-types
    function MockSharedLayout({ children }) {
        return <div className="mock-shared-layout">{children}</div>;
    }
    return MockSharedLayout;
});

// Mock NewLogo component
jest.mock('../../../components/icons/NewLogo', () => {
    // eslint-disable-next-line react/prop-types
    function MockNewLogo({ className }) {
        return <div className={`mock-new-logo ${className || ''}`}>Logo</div>;
    }
    return MockNewLogo;
});

describe('NotFound Component', () => {
    const renderWithHelmet = (component) => {
        return render(<HelmetProvider>{component}</HelmetProvider>);
    };

    test('renders not found page correctly', () => {
        const { container } = renderWithHelmet(<NotFound />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('applies correct CSS classes to container', () => {
        const { container } = renderWithHelmet(<NotFound />);

        const notFoundContainer = container.querySelector('.not-found-container');
        expect(notFoundContainer).toBeInTheDocument();
        expect(notFoundContainer).toHaveClass('not-found-container');
    });

    test('renders logo with correct classes', () => {
        const { container } = renderWithHelmet(<NotFound />);

        const logo = container.querySelector('.mock-new-logo');
        expect(logo).toBeInTheDocument();
        expect(logo).toHaveClass('h-60', 'w-60');
    });

    test('displays 404 title correctly', () => {
        const { getByText } = renderWithHelmet(<NotFound />);

        const title = getByText('404');
        expect(title).toBeInTheDocument();
        expect(title).toHaveClass('title-text-lg');
    });

    test('displays error message correctly', () => {
        const { getByText } = renderWithHelmet(<NotFound />);

        expect(getByText("We couldn't find the page you were looking for.")).toBeInTheDocument();
    });

    test('applies correct CSS classes to text container', () => {
        const { container } = renderWithHelmet(<NotFound />);

        const textContainer = container.querySelector('.not-found-text-container');
        expect(textContainer).toBeInTheDocument();
        expect(textContainer).toHaveClass('not-found-text-container');
    });

    test('renders within SharedLayout', () => {
        const { container } = renderWithHelmet(<NotFound />);

        const sharedLayout = container.querySelector('.mock-shared-layout');
        expect(sharedLayout).toBeInTheDocument();
    });

    test('sets correct page title in helmet', () => {
        renderWithHelmet(<NotFound />);

        // Note: Testing Helmet requires additional setup, but the component structure is tested
        // The Helmet component is rendered and would set the title in a real environment
    });
});
