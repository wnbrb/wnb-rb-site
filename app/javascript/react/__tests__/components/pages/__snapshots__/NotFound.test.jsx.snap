// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component renders not found page correctly 1`] = `
<div
  class="mock-shared-layout"
>
  <div
    class="not-found-container"
  >
    <div
      class="mock-new-logo h-60 w-60"
    >
      Logo
    </div>
    <div
      class="not-found-text-container"
    >
      <h1
        class="title-text-lg"
      >
        404
      </h1>
      <p>
        We couldn't find the page you were looking for.
      </p>
    </div>
  </div>
</div>
`;
